Lab_2_3_4_5/src/Grammar/ParserLR0.java

```diff
@@ -85,55 +87,81 @@ public Map<String, List<ProductionDotIndexTuple>> computeClosureLR0(String sourc
         return closure;
     }
 
-    private void populateActionTable(List<List<ProductionDotIndexTuple>> closure, List<ProductionDotIndexTuple> initialProductions) {
+    private void populateActionTable(List<Map<String, List<ProductionDotIndexTuple>>> closure, Map<String, List<List<String>>>  initialProductionsMap) {
         Integer currentStateIndex = 0;
 
         for(var stateProductions : closure) {
-            if(isAccept(stateProductions)) {
-                actionTable.put("s" + currentStateIndex, "acc");
+            if (isAccept(stateProductions) && isReduce(stateProductions)) {
+                System.out.println("Shift-Reduce conflict detected!");
+                System.out.println(stateProductions);
+            }
+            else if(isAccept(stateProductions)) {
+                actionTable.put(currentStateIndex, "acc");
             } else if(isReduce(stateProductions)) {
-                actionTable.put("s" + currentStateIndex, "reduce " + getInitialProductionIndex(stateProductions.get(0), initialProductions));
+                var onlyKey = stateProductions.keySet().toArray()[0].toString();
+                var production = stateProductions.get(onlyKey).get(0);
+                actionTable.put(currentStateIndex, "reduce " + getInitialProductionIndex(production));
             } else {
-                actionTable.put("s" + currentStateIndex, "shift");
+                actionTable.put(currentStateIndex, "shift");
             }
             currentStateIndex = currentStateIndex + 1;
         }
     }
 
-    private boolean isAccept(List<ProductionDotIndexTuple> stateProductions) {
-        var production = stateProductions.get(0);
+    private boolean isAccept(Map<String, List<ProductionDotIndexTuple>> stateProductions) {
+        var onlyKey = stateProductions.keySet().toArray()[0].toString();
+        var production = stateProductions.get(onlyKey).get(0);
         return  stateProductions.size() == 1 &&
                 production.getProductionSource().compareTo("S'") == 0 &&
                 production.getProductionRhs().size() == 1 &&
                 production.getProductionRhs().get(0).compareTo("S") == 0 &&
                 production.getDotIndex() == 1;
     }
 
-    private boolean isReduce(List<ProductionDotIndexTuple> stateProductions) {
-        var production = stateProductions.get(0);
+    private boolean isReduce(Map<String, List<ProductionDotIndexTuple>> stateProductions) {
+        var onlyKey = stateProductions.keySet().toArray()[0].toString();
```

Comment:
Maybe the name of this variable could have been a little bit more explanatory? sth like "prod_lhs"